service/controller/resource/instance/create_cluster_upgrade_requirement_check.go

```diff
@@ -46,17 +50,38 @@ func (r *Resource) isClusterCreating(cr providerv1alpha1.AzureConfig) bool {
 	return false
 }
 
-func (r *Resource) isClusterScaling(ctx context.Context, cr providerv1alpha1.AzureConfig) (bool, error) {
-	c, err := r.getScaleSetsClient(ctx)
+// anyNodesOutOfDate iterates over all nodes in tenant cluster and finds
+// corresponding azure-operator version from node labels. If node doesn't have
+// this label or was created with older version than currently reconciling one,
+// then this function returns true. Otherwise (including on error) false.
+func (r *Resource) anyNodesOutOfDate(ctx context.Context, cr providerv1alpha1.AzureConfig) (bool, error) {
```

Comment:
CircleCI build is failing due to unused parameter `cr`.

Reply:
Ah, right. Of course. One sec.